sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -41,3 +41,15 @@ select timestamp'2019-10-06 10:11:12.345678' - date'2020-01-01';
 select 3 * (timestamp'2019-10-15 10:11:12.001002' - date'2019-10-15');
 select interval 4 month 2 weeks 3 microseconds * 1.5;
 select (timestamp'2019-10-15' - timestamp'2019-10-14') / 1.5;
+
+-- interval operation with null and zero case
+-- zero division
+select interval '2 seconds' / 0;
+select cast(k as interval) / v from VALUES ('2 seconds', 0) t(k, v);
```

Comment:
My point is that, we shouldn't play tricks in the end-to-end tests to verify the codegen or interpreted code path specially.

If we don't follow this principle, then we need to run all the end-to-end tests twice: once with codegen and once with interpreted.

The end-to-end test should verify the logic, i.e. interval / 0 here. The UT is responsible to make sure codegen and interpreted return the same result.

Reply:
https://github.com/apache/spark/blob/4c53ac18221cbd630bc16352e126d6fc56143239/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/IntervalExpressionsSuite.scala#L227

This doesn't expose the bug, because it needs to convert months to seconds and will hit overflow. Can you try `1 seconds / 0` in that test suite?

Reply:
> My point is that, we shouldn't play tricks in the end-to-end tests to verify the codegen or interpreted code path specially.
> 
> If we don't follow this principle, then we need to run all the end-to-end tests twice: once with codegen and once with interpreted.
> 
> The end-to-end test should verify the logic, i.e. interval / 0 here. The UT is responsible to make sure codegen and interpreted return the same result.

OK，I understand.